components/hubspot/sources/new-event/new-event.js

```diff
@@ -1,77 +1,60 @@
   name: "New Events",
   description: "Emits an event for each new Hubspot event.",
-  version: "0.0.1",
+  version: "0.0.2",
   dedupe: "unique",
   props: {
-    hubspot,
-    objectType: {
-      type: "string",
-      label: "Object Type",
-      optional: false,
-      async options(opts) {
-        return [
-          {
-            label: "Companies",
-            value: "company",
-          },
-          {
-            label: "Contacts",
-            value: "contact",
-          },
-          {
-            label: "Deals",
-            value: "deal",
-          },
-          {
-            label: "Tickets",
-            value: "ticket",
-          },
-        ];
-      },
-    },
+    ...common.props,
+    objectType: { propDefinition: [common.props.hubspot, "objectType"] },
     objectIds: {
-      type: "string[]",
-      label: "Object",
-      optional: false,
-      async options() {
-        let objectType = null;
-        if (this.objectType == "company") objectType = "companies";
-          else objectType = `${this.objectType}s`;
-        const results = await this.hubspot.getObjects(objectType);
-        const options = results.map((result) => {
-          if (objectType == "companies") label = result.properties.name;
-          else if (objectType == "contacts")
-            label = `${result.properties.firstname} ${result.properties.lastname}`;
-          else if (objectType == "deals") label = result.properties.dealname;
-          else if (objectType == "tickets") label = result.properties.subject;
-          return { label, value: JSON.stringify({ label, value: result.id }) };
-        });
-        return options;
-      },
+      propDefinition: [
+        common.props.hubspot,
+        "objectIds",
+        (c) => ({ objectType: c.objectType }),
+      ],
+    },
+  },
+  methods: {
+    ...common.methods,
+    generateMeta(result) {
+      const { id, eventType } = result;
+      return {
+        id,
+        summary: eventType,
+        ts: Date.now(),
+      };
     },
-    db: "$.service.db",
-    timer: {
-      type: "$.interface.timer",
-      default: {
-        intervalSeconds: 60 * 15,
-      },
+    emitEvent(result) {
+      const meta = this.generateMeta(result);
+      this.$emit(result, meta);
+    },
+    isRelevant(result, occurredAfter) {
+      return true;
     },
   },
   async run(event) {
-    const lastRun = this.db.get("occurredAfter") || this.hubspot.monthAgo();
-    const occurredAfter = new Date(lastRun);
+    const occurredAfter =
+      this.db.get("occurredAfter") || Date.parse(this.hubspot.monthAgo());
 
-    const results = await this.hubspot.getEvents(this.objectIds, this.objectType, occurredAfter.getTime());
-    for (const result of results) {
-      this.$emit(result, {
-        id: result.id,
-        summary: `${result.label} ${result.eventType}`,
-        ts: Date.now(),
-      });
+    for (let objectId of this.objectIds) {
+      objectId = JSON.parse(objectId);
```

Comment:
If we're using `objectId.value`, should we just make `this.objectIds` a list of those values instead of a list of objects?

Reply:
Updated